modules/consensus/synchronize.go

```diff
@@ -76,13 +87,26 @@ func (cs *ConsensusSet) threadedReceiveBlocks(conn modules.PeerConn) error {
 		return err
 	}
 
+	// Broadcast the last block accepted. This functionality is in a defer to
+	// ensure that a block is always broadcast if any blocks are accepted. This
+	// is to stop an attacker from preventing block broadcasts.
+	chainExtended := false
+	defer func() {
+		if chainExtended {
+			// The last block received will be the current block since
+			// managedAcceptBlock only returns nil if a block extends the longest chain.
+			currentBlock := cs.CurrentBlock()
```

Comment:
no, you only want to annoy your peers if you've found a longer chain that they might not know about yet. If you don't get any extending blocks, there's no reason to tell your peers because they already know about the longest chain (either you broadcast it to them when you got the longest block, or they asked you about it when they connected to you).